tests/integration/framework/BUILD

```diff
@@ -0,0 +1,14 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "component.go",
```

Comment:
Perhaps break this and create multiple libraries?
component is needed for creating components to extend an environment.
environment is needed for assembling different components.
framework is needed to run tests